### PR TITLE
Html out  - working version

### PR DESCRIPTION
**Pull request by haim-kermany**
this is a PR based on Shmulik work (added to master suing cherry pick) and minimal changes to make it work.
still needs to:
1. figure out where to activate the explain tracker when using scheme file
2. figure out why we output xml data in text files output
3. add relevant tests

### CODE REVIEW

File: nca/nca_cli.py [hunk 1]
```diff
@@ -187,8 +187,11 @@ def run_args(args):  # noqa: C901
 
     if args.explain is not None and args.optimized_run == 'true':
         output_config['explain'] = args.explain
-        ExplTracker().activate()
-        ExplTracker().set_endpoints(output_config.outputEndpoints)
+        ExplTracker().activate(output_config.outputEndpoints)
+
+    if args.output_format == 'html':
```
Comment (adisos) [inline nca/nca_cli.py:192]: stop the run if optimized flag is not true?

File: nca/FWRules/InteractiveConnectivityGraph.py [hunk 2]
```diff
@@ -59,13 +61,13 @@ class ElementRelations:
         highlights: set = field(default_factory=set)
         explanation: list = field(default_factory=set)
 
-    def __init__(self, svg_file_name, output_directory):
+    def __init__(self, svg_file_name, output_directory, expl_xml=None):
```
Comment (adisos) [inline nca/FWRules/InteractiveConnectivityGraph.py:64]: update `init` params in the documentation

File: nca/FWRules/InteractiveConnectivityGraph.py [hunk 3]
```diff
@@ -111,7 +113,7 @@ class SvgGraph:
 
         ELEMENTS_DIRECTORY = 'elements'
 
-        def __init__(self, input_svg_file, output_directory):
+        def __init__(self, input_svg_file, output_directory, expl_xml=None):
```
Comment (adisos) [inline nca/FWRules/InteractiveConnectivityGraph.py:116]: same here

File: nca/nca_cli.py [hunk 4]
```diff
@@ -322,7 +325,7 @@ def nca_main(argv=None):
                         help='A list of labels to subset the query by')
     parser.add_argument('--ghe_token', '--gh_token', type=str, help='A valid token to access a GitHub repository')
     parser.add_argument('--output_format', '-o', type=str,
-                        help='Output format specification (txt, txt_no_fw_rules, csv, md, dot, jpg or yaml). '
+                        help='Output format specification (txt, txt_no_fw_rules, csv, md, dot, jpg, html or yaml). '
```
[the remark this answers is not in the document]
Comment (haim-kermany) [reply nca/nca_cli.py:328]: done

File: nca/nca_cli.py (hunk 1 above)
Comment (haim-kermany) [reply nca/nca_cli.py:192]: fixed

File: nca/FWRules/InteractiveConnectivityGraph.py (hunk 2 above)
Comment (haim-kermany) [reply nca/FWRules/InteractiveConnectivityGraph.py:64]: done

File: nca/FWRules/InteractiveConnectivityGraph.py (hunk 3 above)
Comment (haim-kermany) [reply nca/FWRules/InteractiveConnectivityGraph.py:116]: done

Comment (haim-kermany) [reply nca/FWRules/InteractiveConnectivityGraph.py:117]: see task for nest PR at:
https://github.com/IBM/network-config-analyzer/issues/662

**Review by adisos — APPROVED** [review]

**Pull request merged**